### ISSUE TITLE
Segment replication and recovery contend and share limits

### ISSUE THREAD

**Issue by andrross** [opensearch-project/OpenSearch#10624]
Segment replication (both node-to-node and the remote store variants) shares the same code paths as shard recovery, and therefore honors the same limits, such as `indices.recovery.max_bytes_per_sec`. At some level they are using the same physical resources (network bandwidth, disk bandwidth, etc), so some amount of contention is inevitable. However, they are presently not independently tunable or prioritizable. Throttling recovery often makes sense, because you don't want to impact other operations of a cluster to reallocate a shard or allocation a new replica, for example. However, steady-state replication seems more "business critical" so there may well be cases where replication might want to be limited differently than recovery.

**Comment by Bukhtawar** [comment]
Thanks @andrross for opening this. @gbbafna had raised it too while we integrated rate limiters 
Curious did you notice the contention in any of your interaction

**Comment by andrross** [comment]
@Bukhtawar Yes. With the introduction of parallel file download, there is a thread pool with a max size of "half processor count max 10". Each recovery and replication operation kicks off a configurable number of concurrent downloads. With some combination of a small host (like 2 or 4 CPUs), a low `max_bytes_per_sec` limit, or a lot of multi-GB segment files, then it is pretty easy to saturate this thread pool and cause replication events to queue up.
